Additional holidays and holiday fix for Australia

arguellesjonas opened the pull request:
NT / Darwin
        Sunday 20 April - Easter Sunday
        Monday 4 August - Picnic Day
    Queensland / Brisbane
        Sunday 20 April - Easter Sunday
    South Australia / Adelaide
        Sunday 20 April - Easter Sunday
    Western Australia / Perth
        Sunday 20 April - Easter Sunday

arguellesjonas committed:
Additional holiday fix
    NT / Darwin
        Sunday 20 April - Easter Sunday
        Monday 4 August - Picnic Day
    Queensland / Brisbane
        Sunday 20 April - Easter Sunday
    South Australia / Adelaide
        Sunday 20 April - Easter Sunday
        Friday 25 April <= this date is listed twice
    Western Australia / Perth
        Sunday 20 April - Easter Sunday
        Fri - 25 Apr 2025 <= this date is listed twice

kcsuyog commented on definitions/au.yaml:
```diff
@@ -54,9 +54,9 @@ months:
-    regions: [au]
-    mday: 25
+  - name: Easter Sunday
+    regions: [au_nt, au_qld, au_sa, au_wa]
```
@arguellesjonas can you confirm we are not suposed to add easter sunday for nsw?

kcsuyog replied:
regions: [au]

kcsuyog replied:
with method saying , enable this from 2025

kcsuyog replied:
```
0
  - name: Easter Sunday
    regions: [au_nt, au_qld, au_sa, au_wa]
    function: easter_from_2025

- easter_from_2025: 
    arguments: year
    source: |
      if year >=2025
        easter(date)
      end
```

kcsuyog replied:
4:
  - name: Easter Sunday
    regions: [au_nt, au_qld, au_sa, au_wa]
    function: easter_sunday_for_2025

arguellesjonas replied:
This is done now

kcsuyog commented on definitions/au.yaml:
```diff
@@ -54,9 +54,9 @@ months:
-    mday: 25
+  - name: Easter Sunday
+    regions: [au_nt, au_qld, au_sa, au_wa]
+    mday: 20
```
revert this

kcsuyog commented on definitions/au.yaml:
```diff
@@ -105,6 +105,9 @@ months:
     regions: [au_qld_brisbane]
     week: -3
     wday: 3
+  - name: Picnic Day
```
needs to have a function to enable this from 2025 onwards

arguellesjonas replied:
This is done now.

arguellesjonas committed:
PR request changes: Add functions for 2025 holiday for Picnic and easter

arguellesjonas committed:
Fix deprecation warning to use Date#to_fs

arguellesjonas committed:
Fix Easter Sunday implementation for 2025

arguellesjonas committed:
Revert ANZAC Day

arguellesjonas committed:
Revert generation changes for ANZAC

arguellesjonas committed:
Revert deprecation fix for Date#to_fs

kcsuyog commented on definitions/au.yaml:
```diff
@@ -173,6 +179,12 @@ methods:
+    arguments: year
+    source: |
+      if year >= 2025
+        DateCalculatorFactory::Easter::Gregorian.easter_calculator.calculate_easter_for(year)
```
didnt realized we had this method

tob1k committed:
Avoid implicitly calling deprecated Date#to_s conversion

tob1k committed:
Merge pull request #12 from tob1k/holidays-april-16-2025-patched

Fix Ruby 3.1+ Date#to_s deprecation warning in ProcResultCache

mikkotansc committed:
Add ANZAC Day 2026 observed holiday for NSW (April 27)

ANZAC Day falls on Saturday April 25, 2026. NSW has announced
Monday April 27 as an additional public holiday for 2026 only.

mikkotansc committed:
Add ANZAC Day 2027 observed holiday for NSW (April 26)

ANZAC Day falls on Sunday April 25, 2027. NSW observes Monday
April 26 as an additional public holiday.

mikkotansc committed:
Merge pull request #13 from shiftcare/feature/add-april-27-2026-holiday

Add ANZAC Day 2026 observed holiday for NSW (April 27)
